modules/issue_tracker/ajax/EditIssue.php

```diff
@@ -179,13 +180,60 @@ function validateInput($values)
     $db         =& Database::singleton();
     $pscid      = (isset($values['PSCID']) ? $values['PSCID'] : null);
     $visitLabel = (isset($values['visitLabel']) ? $values['visitLabel'] : null);
+    $centerID   = (isset($values['centerID']) ? $values['centerID'] : null);
     $result     = [
-                   'PSCID'     => $pscid,
-                   'visit'     => $visitLabel,
-                   'candID'    => null,
-                   'sessionID' => null,
+                   'PSCID'             => $pscid,
+                   'visit'             => $visitLabel,
+                   'centerID'          => $centerID,
+                   'candID'            => null,
+                   'sessionID'         => null,
+                   'isValidSubmission' => true,
+                   'invalidMessage'    => null,
                   ];
 
+    if (isset($result['PSCID'], $result['centerID'])) {
+        $valid_center = $db->pselectOne(
+            "
+            SELECT
+                CenterID = :center_id
+            FROM
+                candidate
+            WHERE
+                PSCID = :psc_id
+        ",
+            array(
+             "center_id" => $result['centerID'],
+             "psc_id"    => $result['PSCID'],
+            )
+        );
+        if (!$valid_center) {
+            $valid_center = $db->pselectOne(
+                "
+                SELECT
```

Comment:
A candidate is initially given a `CenterID`, yes. However, the candidate could potentially visit more than one site. So, if the `CenterID` references `Montreal` but the candidate visited `Ottawa`... And we want to open a ticket related to the `Ottawa` visit, we'd need to allow that, is my opinion, at least.

Without this extra check, you'll only ever be able to specify *just one site per candidate*. At that point, we might as well not allow the user to ever specify a site and just derive the site from the candidate, since it'll have a one-to-one mapping.